### PR TITLE
add fields and adjust family structure

### PR DESCRIPTION
**Pull request by rosibaj**
- Added diagnosis_category and affected_status

- Changes family structure to not nested; should be the same for all participants in a family

### CODE REVIEW

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+            }, 
+            "diagnoses": {
+              "properties": {
+                "age_at_event_days": {
+                  "type": "integer"
+                }, 
+                "diagnosis": {
+                  "type": "keyword"
+                }, 
+                "diagnosis_category": {
```
Comment (junjun-zhang) [inline es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:130]: `diagnosis_category` is not defined in `es_mapper.yaml`, that means it should be in `dataservice` API.

The question is: `diagnosis_category` appears in both under `diagnoses` and `participant`, is this the case in `dataservice` api? Please verify.

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+            }, 
+            "consent_type": {
+              "type": "keyword"
+            }, 
+            "diagnoses": {
+              "properties": {
+                "age_at_event_days": {
+                  "type": "integer"
+                }, 
+                "diagnosis": {
```
Comment (junjun-zhang) [inline es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:127]: This is defined in `es_mapper.yaml` but not in dataservice api, should this be removed?

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+            }, 
+            "ethnicity": {
+              "type": "keyword"
+            }, 
+            "external_id": {
+              "type": "keyword"
+            }, 
+            "family": {
+              "properties": {
+                "available_data_types": {
```
Comment (junjun-zhang) [inline es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:174]: Are fields directly under `family` supposed to be taken from `dataservice` api? If that's the intention, there are a few fields need to be removed from `es_mapper.yaml`, such as: `available_data_types`, `shared_hpo_ids`, `father_id` and `mother_id`. These fields were added by the ETL previously.

If some of these fields are still intended to be populated by the ETL, then keep those.

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+                }, 
+                "family_members": {
+                  "properties": {
+                    "affected_status": {
+                      "type": "boolean"
+                    }, 
+                    "alias_group": {
+                      "type": "keyword"
+                    }, 
+                    "available_data_types": {
```
Comment (junjun-zhang) [inline es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:188]: similar question as a previous one, `available_data_types` for a family member is not from the participant entity of the `dataservice`, keep the field only when ETL will populate the value of it.

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json (hunk 3 above)
Comment (rosibaj) [reply es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:174]: We had not discussed removing those so I left them in to continue to be populated by ETL - for example `father_id` and `mother_id` are in use in the reports.

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json (hunk 2 above)
Comment (rosibaj) [reply es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:127]: This was to support this field requested by CHOP:
https://kidsfirst.atlassian.net/wiki/spaces/KFDRC/pages/748027905/ETL+Calculations#ETLCalculations-Diagnosis

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json (hunk 1 above)
Comment (rosibaj) [reply es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:130]: @junjun-zhang yes it appears in both places - each `diagnosis` has a `diagnosis_category` of cancer or structural birth defect.  The `diagnosis_category` under `participant` is an aggregated rollup of the individual diagnosis `diagnosis_category` into a single classification of birth defect, cancer, or birth defect and cancer

File: es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json (hunk 2 above)
Comment (junjun-zhang) [reply es-model-archive/kf-es-model-v0.5.0_1.2.0/file_centric.mapping.json:127]: If this is populated by the ETL, then yes, we keep it.

**Review by junjun-zhang — APPROVED** [review]
looks good

**Pull request merged**